src/core/service.c

```diff
@@ -1982,7 +1982,7 @@ static void service_enter_dead(Service *s, ServiceResult f, bool allow_restart)
         if (unit_stop_pending(UNIT(s)))
                 allow_restart = false;
 
-        if (s->result == SERVICE_SUCCESS)
+        if (f == SERVICE_SUCCESS || s->result == SERVICE_SUCCESS)
```

Comment:
For services, we go through `stop-post` state on failure, and if that succeeds the result would be reset to success? This would be quite confusing?